fix(ivy): support multiple exportAs

crisbeto opened the pull request:
Allows for multiple, comma-separated `exportAs` names, similarly to `ViewEngine`.

These changes fix FW-708.

mary-poppins commented:
You can preview 7b89b3e at https://pr27996-7b89b3e.ngbuilds.io/.

mary-poppins commented:
You can preview d97ab1f at https://pr27996-d97ab1f.ngbuilds.io/.

mary-poppins commented:
You can preview f828c7f at https://pr27996-f828c7f.ngbuilds.io/.

mary-poppins commented:
You can preview 90d6a83 at https://pr27996-90d6a83.ngbuilds.io/.

alxhub commented on packages/compiler/src/render3/view/compiler.ts:
```diff
@@ -117,7 +117,8 @@ function baseDirectiveFields(
 
   if (meta.exportAs !== null) {
-    definitionMap.set('exportAs', o.literal(meta.exportAs));
+    const exportAsValues = meta.exportAs.split(',').map(part => o.literal(part.trim()));
```
`meta.exportAs` should already be split here. We should do that processing for AOT in [ngtsc/annotations/src/directive.ts](https://github.com/angular/angular/blob/a0840242d7cf695a4bc41d39a6f937a72e6bd722/packages/compiler-cli/src/ngtsc/annotations/src/directive.ts#L178) and for JIT in [render3/jit/directive.ts](https://github.com/angular/angular/blob/a0840242d7cf695a4bc41d39a6f937a72e6bd722/packages/core/src/render3/jit/directive.ts#L153).

For the compiler, it's actually a little more complicated as `exportAs` is one of the metadata elements written to `.d.ts` files, so that needs to change as well. Additionally there's the template type-checking code which matches against this...

Let me make you a precursor PR for the compiler which adds this support that you can build on.

mary-poppins commented:
You can preview d021ced at https://pr27996-d021ced.ngbuilds.io/.

crisbeto committed:
fix(ivy): support multiple exportAs

Allows for multiple, comma-separated `exportAs` names, similarly to `ViewEngine`.

These changes fix FW-708.

kara commented:
[presubmit](http://test/OCL:228767245:BASE:228767272:1547156050182:60833465)

alxhub commented:
> core/todo/bundle increased by 41 bytes.

Unacceptable!! ;)
